doc/kernel/other/ring_buffers.rst

```diff
@@ -114,6 +155,15 @@ which can be accessed using less efficient modulo operations.
     #define MY_RING_BUF_WORDS 93
     SYS_RING_BUF_DECLARE_SIZE(my_ring_buf, MY_RING_BUF_WORDS);
 
+The following code defines a ring buffer with an arbitrary-sized data buffer,
```

Comment:
How about instead:

    The following code defines an application-specific sized **byte mode** ring buffer
    enqueued and dequeued as raw bytes: